data/types/blood/glucose/glucose.go

```diff
@@ -33,3 +36,75 @@ func (g *Glucose) Normalize(normalizer data.Normalizer) {
 		g.Value = dataBloodGlucose.NormalizeValueForUnits(g.Value, units)
 	}
 }
+
+func roundToEvenWithDecimalPlaces(v float64, decimals int) float64 {
+	if decimals < 1 {
+		return math.RoundToEven(v)
+	}
+	coef := math.Pow(10, float64(decimals))
+	return math.RoundToEven(v*coef) / coef
+}
+
+type Classification string
+
+const (
+	ClassificationInvalid Classification = "invalid"
+
+	VeryLow       = "very low"
+	Low           = "low"
+	OnTarget      = "on target"
+	High          = "high"
+	VeryHigh      = "very high"
+	ExtremelyHigh = "extremely high"
+)
+
+type classificationThreshold struct {
+	Name      Classification
+	Value     float64
+	Inclusive bool
+}
+
+type Classifier []classificationThreshold
+
+func (c Classifier) Classify(g *Glucose) (Classification, error) {
+	normalized, err := dataBloodGlucose.NormalizeValueForUnitsSafer(g.Value, g.Units)
+	if err != nil {
+		return ClassificationInvalid, errors.Wrap(err, "unable to classify")
+	}
+	// Rounded values are used for all classifications. To not do so risks introducing
+	// inconsistency between frontend, backend, and/or other reports. See BACK-3800 for
+	// details.
+	rounded := roundToEvenWithDecimalPlaces(normalized, 1)
+	for _, threshold := range c {
+		if threshold.Includes(rounded) {
+			return threshold.Name, nil
+		}
+	}
+	// Ensure your highest threshold has a value like math.MaxFloat64 to avoid this.
+	return ClassificationInvalid, errors.Newf("unable to classify value: %v", *g)
+}
+
+// TidepoolADAClassificationThresholdsMmolL for classifying glucose values.
+//
+// All values are normalized to MmolL before classification.
+//
+// In addition to the standard ADA ranges, the Tidepool-specifiic "extremely high" range is
+// added.
+//
+// It is the author's responsibility to ensure the thresholds remain sorted from smallest to
+// largest.
+var TidepoolADAClassificationThresholdsMmolL = Classifier([]classificationThreshold{
```

Comment:
Hmm, including the `Name` in the `classificationThreshold` could lead to some future errors. For example, not realizing that the order was important, someone could reorder these and cause havoc. Consider a map and explicitly classifying in the correct order (i.e. [VeryLow, Low, InRange, High, VeryHigh] and anything else is ExtremelyHigh)?

Reply:
Oh, and if a map doesn't include a classification (e.g. 'Low') skip it and go to the next one (e.g. 'In Range').

Reply:
I'll take a look at what I can do. Using a map doesn't prevent errors (the user could miss a key, or leave a key out of the explicit ordering), and I don't wanna overcomplicate it too much.

If someone doesn't read the comment *and* misses the fact that it's using an ordered data structure *and* ignores all the unit tests *and* gets the change past code review... Forgive me, but that's on them, don't you think? There are limits to what I can prevent. I feel like I've taken reasonable steps.

Reply:
I've applied a sort to the thresholds. I dunno that its worth it, but... it's... something. Let me know what you think.